frontend/src/components/Searchbar/Searchbar.tsx

```diff
@@ -24,21 +25,44 @@ export interface SearchbarProps extends InputProps {
   onSearch: (searchValue: string) => void
 
   /**
-   * Whether the searchbar is expanded or not.
-   * @note This should be `true` if the `onSearchIconClick` function prop is not
-   * provided, or the searchbar will not be usable.
+   * Whether the searchbar is expandable/collapsable or not. Defaults to `true`.
    */
-  isExpanded: boolean
+  isExpandable?: boolean
 
   /**
-   * Optional. Function to be invoked when the search icon is clicked.
-   * If provided, the search icon will be clickable.
+   * Whether the searchbar is initially expanded or not. Defaults to `false`.
+   * @note Set this to `true` if `isExpandable` is set to `false`, else the
+   * searchbar will not be usable.
    */
-  onSearchIconClick?: () => void
+  isInitiallyExpanded?: boolean
+
+  /**
+   * Called when the search icon is clicked and the search bar is expanded.
+   */
+  onExpand?: () => void
+
+  /**
+   * Called when the X icon is clicked and the search bar is collapsed.
+   */
+  onCollapse?: () => void
 }
 
 export const Searchbar = forwardRef<SearchbarProps, 'input'>(
-  ({ onSearch, isExpanded, onSearchIconClick, ...props }, ref) => {
+  (
+    {
+      onSearch,
+      isExpandable = true,
+      isInitiallyExpanded = false,
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
+      onExpand,
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
+      onCollapse,
+      ...props
+    },
+    ref,
+  ) => {
+    const [isExpanded, setIsExpanded] = useState<boolean>(isInitiallyExpanded)
```

Comment:
In, relation to my comment above, to show the priority and fallback, this could possibly be 
```suggestion
    const [isActuallyExpanded, setIsExpanded] = useState<boolean>(!isExpandable || isExpanded)
```

Again, I realize isActuallyExpanded sounds weird. Perhaps, we can use a simple prefix to denote the internal state variable:
```suggestion
    const [_isExpanded, setIsExpanded] = useState<boolean>(!isExpandable || isExpanded)
```

Reply:
updated: `isNowExpanded`